src/LineUI/LineUnit.tsx

```diff
@@ -165,11 +176,23 @@ const LineUnit : React.FC<ILineUnitProps> = (props) => {
       {/* <circle r={1* unit} cx={x1} cy={y1} fill='white' /> */}
       {/* <circle r={1* unit} cx={x2} cy={y2} fill='white' /> */}
 
-      {
-        label &&
-          <LabelText styling={styling} fontSize={`${unit * 14}px`} x={midpoint.x - (16 * unit)} y={midpoint.y - (15 * unit)} showIndex={revealSetIndex || handleFinderActive}>
-            {label}
-          </LabelText>
+      {label &&
+        <LabelText
+          styling={styling}
+          fontSize={`${unit * 14}px`}
+          x={midpoint.x - ((centerIconPosition === 'up' ? 10 : 20) * unit) }
+          y={midpoint.y  + (centerIconPosition === 'up' ? (30 * unit) : -(15 * unit))}
+          showIndex={revealSetIndex || handleFinderActive}
+        >
+          {label}
+        </LabelText>}
+      
+      {centerIconPosition && CenterIcon &&
+        <g transform={`translate(${centerIconCordinates().x},${centerIconCordinates().y}) rotate(${(180 / Math.PI) * Math.atan2(y2 - y1, x2 - x1) + (centerIconPosition === 'up' ? 0 : 180)}) scale(${unit * 1.5})`} >
+          <g transform={centerIconPosition === 'up' ? `translate(${Math.ceil(4-unit) > 2 ? 10 : 5 - unit}, -30)` : `translate(${(-25 + 1.5 * unit) < -21 ? -28 : -25 + 1.5 * unit}, -30)`}>
+            <CenterIcon height='20' width='20' />
+          </g>
+        </g>
```

Comment:
Related to the comments on dropping reliance on `UP` or `DOWN`, this can be simplified.